typed-function.js

```diff
@@ -1123,8 +1124,21 @@
         // so it's fine to have unnamed functions.
       }
 
-      // attach signatures to the function
+      // attach signatures to the function.
+      // This first property is close to the original collection of signatures
+      // used to create the typed-function, just with unions split:
       fn.signatures = createSignaturesMap(signatures);
+      // Whereas the signaturesArray property contains the exact list of
+      // tests and implementing functions with possible conversions compiled
+      // in, that are used in executing the typed-function
+      for (let i = 0; i < signatures.length; ++i) {
+        signatures[i].implementation = fns[i]
+        signatures[i].test = tests[i]
+      }
+      fn.signatureArray = signatures
```

Comment:
Thinking aloud here: if possible I would like to keep the API concise and not offer multiple "things" with more or less the same. Would it be ok to only expose `signatures`, and have `resolve` iterate over the object instead of over an array?

Reply:
Well the problem is they have different info. maybe just the naming is bad. signatures has just the original specified signatures (albeit split) which is what you need for merging more signatures. And since it is not specific enough,, it has no tests or implementations. signatureArray has every detailed possibility in resolution order with tests and implementations -- everything you need to actually make the typed-function work. It is conceivable one could recover the current contents of signatures from signatureArray but I wasn't sure and it didn't seem like it was worth the thought and programming effort to try.

Frankly I think what I called 'signatureArray' should be considered a "private" implementation detail, whereas signatures is more a part of the api. So perhaps the new property should just be renamed 'implementations' or even '_implementations' to flag this?

Reply:
> Well the problem is they have different info.

Ah of course. I'm sometimes getting confused by the naming 😂 , also on whether "signature" is a string or the actual function for example.

Then it is indeed needed to expose this information. We could indeed use a prefix `_`, and name it something like `_implementations`. Or, how about `_parsedSignatures`? 

Thinking aloud here: we see this information as "private". Maybe instead of exposing this information, we should expose the public method we want to use it for: expose `f.resolve(argList)` as a method on the typed function itself, instead as a static function `typed.resolve(f, argList)`?

Reply:
OK, will put resolve on the typed function itself and then the "signatureArray" will be inaccessible, making its name irrelevant.

Reply:
Nice, l like it 👌